build/lib/release.sh

```diff
@@ -115,7 +115,7 @@ function kube::release::package_src_tarball() {
           \) -prune \
         \))
     )
-    "${TAR}" czf "${src_tarball}" -C "${KUBE_ROOT}" "${source_files[@]}"
+    "${TAR}" czf "${src_tarball}" --transform 's|^\.|kubernetes|' -C "${KUBE_ROOT}" "${source_files[@]}"
```

Comment:
Would this be safer? I'm assuming this is what this does.

```suggestion
    "${TAR}" czf "${src_tarball}" --transform 's|^\./|kubernetes/|' -C "${KUBE_ROOT}" "${source_files[@]}"
```

Reply:
@cblecker the original code works fine. from my trial and error the second one adds an extra `'/'` i believe. we should be ok with what we had originally

Reply:
I think we needed to ensure gnu tar here https://github.com/kubernetes/kubernetes/issues/99818

Reply:
actually confusingly TAR env should be gnu tar so ... investigating in that issue.